src/main/java/com/thevoxelbox/voxelsniper/util/io/FileUtils.java

```diff
@@ -0,0 +1,54 @@
+package com.thevoxelbox.voxelsniper.util.io;
+
+import com.sk89q.worldedit.internal.util.LogManagerCompat;
+import com.thevoxelbox.voxelsniper.VoxelSniperPlugin;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class FileUtils {
+
+    private static final Logger LOGGER = LogManagerCompat.getLogger();
+
+    public static File copyFile(String resource, File output) {
+        return copyFile(resource, output, resource);
+    }
+
+    public static File copyFile(String resource, File output, String fileName) {
+        try {
+            if (output == null) {
+                output = VoxelSniperPlugin.plugin.getDataFolder();
+            }
+            if (!output.exists()) {
+                output.mkdirs();
+            }
+            File newFile = new File(output, fileName);
+            if (newFile.exists()) {
+                return newFile;
+            }
+            try (InputStream stream = VoxelSniperPlugin.class.getResourceAsStream(resource.startsWith("/") ? resource :
+                    "/" + resource)) {
+                byte[] buffer = new byte[2048];
```

Comment:
Instead of the manual buffer read/write stuff, you can just use `Files.newOutputStream(newFile.toPath())` (add it as resource to the try-with-resources statement) and then just call `stream.transferTo(newFileStream)`.

Note that `stream` can be null, maybe it's worth to throw an appropriate exception to avoid confusing NPEs.

Reply:
Thanks. _Such fixes should be also rectified on FAWE then 🤔._

Reply:
Indeed. Feel free to create a PR there too if you want :)